### PR TITLE
src/test/cli/* tests: POSIX Convert grep -P to grep -E

### PR DESCRIPTION
**Pull request by wjwithagen**
- FreeBSD does not have grep -P

Signed-off-by: Willem Jan Withagen <wjw@digiware.nl>

### CODE REVIEW

File: src/test/cli/ceph-authtool/cap-overwrite.t [hunk 1]
```diff
@@ -2,10 +2,10 @@
   creating kring
 
   $ ceph-authtool --cap osd 'allow rx pool=swimming' kring
-  $ ceph-authtool kring --list|grep -P '^\tcaps '
+  $ ceph-authtool kring --list|grep -E '^\Wcaps '
```
Comment (tchaikov) [inline src/test/cli/ceph-authtool/cap-overwrite.t]: i don't think it's necessary to loosen the check for tab.

**Comment by wjwithagen** [comment]
@tchaikov 
This more to your liking?

File: src/test/cli/ceph-authtool/cap-invalid.t [hunk 2]
```diff
@@ -3,10 +3,10 @@
 
 # TODO is this nice?
   $ ceph-authtool --cap osd 'broken' kring
-  $ ceph-authtool kring --list|grep -P '^\tcaps '
+  $ ceph-authtool kring --list|grep -E '^[^I]caps '
```
Comment (tchaikov) [inline src/test/cli/ceph-authtool/cap-invalid.t]: i never knew that ERE supported the syntax like this! thought it's a character class excluding capital "I"! and seems the spec does not mention this usage, see http://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap09.html#tag_09_03_05 .

Comment (wjwithagen) [inline src/test/cli/ceph-authtool/cap-invalid.t]: On 20-7-2016 12:37, Kefu Chai wrote:
> In src/test/cli/ceph-authtool/cap-invalid.t
> <https://github.com/ceph/ceph/pull/10319#discussion_r71502243>:
> 
>> @@ -3,10 +3,10 @@
>>  
>>  # TODO is this nice?
>>    $ ceph-authtool --cap osd 'broken' kring
>> -  $ ceph-authtool kring --list|grep -P '^\tcaps '
>> +  $ ceph-authtool kring --list|grep -E '^[^I]caps '
> 
> i never knew that ERE supported the syntax like this! thought it's a
> character class excluding capital "I"! and seems the spec does not
> mention this usage, see
> http://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap09.html#tag_09_03_05

It is one of the few versions that I actually got to work with grep -E
Took me a while to get something that works on all ends and pieces.
So I'm very curious what a teuthology runs will bring.

Comment (tchaikov) [inline src/test/cli/ceph-authtool/cap-invalid.t]: yeah, it works for me, anyway.

**Comment by tchaikov** [comment]
lgtm.

**Comment by wjwithagen** [comment]
On 20-7-2016 13:01, Kefu Chai wrote:
> lgtm.

Looked over it again, and it is definitly not correct. :(
Replace ^I with ^K and it still works.
Back to the drawing board, because ERE does not seem to take the
standard escapes.

So I'm very tempted to use [[:space:]], that would allow either
<space> of <tab>.
So less liberal than \W

**Comment by wjwithagen** [comment]
@tchaikov 
Jenkins likes this too, and that is really according the Posix standard.
Catches blanks, and some other spacing stuff. But not too liberal.

**Comment by tchaikov** [comment]
tested at

- http://pulpito.ceph.com/kchai-2016-07-24_02:17:13-rados-wip-kefu-testing---basic-mira/
- http://pulpito.ceph.com/kchai-2016-07-24_20:41:33-rados-wip-kefu-testing---basic-mira/

**Comment by wjwithagen** [comment]
@tchaikov 
Thanx

**Pull request merged**